web/src/components/CustomMarquee/CustomMarquee.jsx

```diff
@@ -0,0 +1,75 @@
+import React from "react";
+import Marquee from "react-fast-marquee";
+
+import PropTypes from "prop-types";
+
+function CustomMarquee(props) {
+    const {
+        data,
+        containerRef,
+    } = props;
+
+    const windowSize = React.useRef(window.innerWidth);
+    const resizeCooldown = React.useRef(false);
+
+    const dataRef = React.useRef(null);
```

Comment:
where do you set this?

Reply:
I'm using refs as a document.getElementById, which you can't use directly because react elements are too transient to have an ID at all times
https://react.dev/learn/manipulating-the-dom-with-refs#getting-a-ref-to-the-node

You send a null useRef to the value of a ref prop on an element and it magics it together to be a functional equivalent to document.getElementById for me to read the innerWidth of

Reply:
where do you set this tho? i don't see a `dataRef.current = [...]` anywhere

Reply:
I suppose that link wasn't a perfect example

You don't need to set the value of `current` directly at any point in some cases, this being one of them

I set the value of that ref in the div inside of the Marquee, and I use that to measure the width of the content vs the width of the marquee to decide if to should or shouldn't scroll

```

        <Marquee play={dataScroll} speed={30} onCycleComplete={() => {setOverride()}}>
            <div
                style={{marginLeft: "10px"}}
                className="marquee-text"
                ref={dataRef} // Here
            >
                {data}
            </div>
        </Marquee>
```

And then during assessMarquee(), I set the value of a const to the value of dataRef.current so I can use that ref as a reference to the element and measure the relative widths